pkg/apis/feeds/v1alpha1/common_event_source_types.go

```diff
@@ -40,6 +41,10 @@ type CommonEventSourceSpec struct {
 	Parameters *runtime.RawExtension `json:"parameters,omitempty"`
 }
 
+// Check that CommonEventSourceSpec can be validated and can be defaulted
+var _ apis.Validatable = (*CommonEventSourceSpec)(nil)
+var _ apis.Defaultable = (*CommonEventSourceSpec)(nil)
```

Comment:
You don't want to validate immutability? Is that because Immutability is only a top-level check?

Reply:
I did not plumb down the immutability to the common specs.